core/store/models/user.go

```diff
@@ -71,3 +76,45 @@ type ChangePasswordRequest struct {
 	OldPassword string `json:"oldPassword"`
 	NewPassword string `json:"newPassword"`
 }
+
+// ChangeAuthTokenRequest is sent when updating a User's authentication token.
+type ChangeAuthTokenRequest struct {
+	Password string `json:"password"`
+}
+
+// GenerateAuthToken randomly generates and sets the users Authentication
+// Token.
+func (u *User) GenerateAuthToken() (*AuthToken, error) {
+	token := NewAuthToken()
+	return token, u.SetAuthToken(token)
+}
+
+// DeleteAuthToken clears and disables the users Authentication Token.
+func (u *User) DeleteAuthToken() {
+	u.TokenKey = ""
+	u.TokenSalt = ""
+	u.TokenHashedSecret = ""
+}
+
+// SetAuthToken updates the user to use the given Authentication Token.
+func (u *User) SetAuthToken(token *AuthToken) error {
+	salt := utils.NewSecret(utils.DefaultSecretSize)
+	hashedSecret, err := HashedSecret(token, salt)
+	if err != nil {
+		return errors.Wrap(err, "user")
+	}
+	u.TokenSalt = salt
+	u.TokenKey = token.AccessKey
+	u.TokenHashedSecret = hashedSecret
+	return nil
+}
+
+// AuthenticateUserByToken returns true on successful authentication of the
+// user against the given Authentication Token.
+func AuthenticateUserByToken(token *AuthToken, user *User) (bool, error) {
```

Comment:
Idea: Make instance method on `user *User`. Why?
1. More discoverable API
2. Groups it along with the rest of the user Token methods

Reply:
This makes sense though it was implemented this way to be consistent with the external initiator's authentication method.

https://github.com/smartcontractkit/chainlink/blob/develop/core/store/models/external_initiator.go#L59